### PR TITLE
feat: Merge path hostname input box

### LINKED ISSUE #733: Operation experience optimization
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [ ] Bug
- [ ] Requirements
- [x] Feature or performance improvement
- [ ] Other

___
### Requirements or improvement
image
It is recommended to merge the two input to input URL instead,you don't have to worry about HTTP/HTTPS or IP/Domain or various query strings.

### PR DESCRIPTION
**Pull request by Si-ege**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.

**Related issues**

fix/resolve #733
1.Merge path  hostname input box.
2.Splitting paths and hostnames before sending requests.

**Checklist:**

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by zaunist** [comment]
Hi, @Si-ege . Could you please provide your PR related issues? Or give the context? I don't know why we need this change.

**Comment by zaunist** [comment]
Hi, @Si-ege . Could you remove "()" from title ? I think it's a little strange.

**Comment by guoqqqi** [comment]
Hi @Si-ege I see that some of the changes were made by #2327, are you developing on the same branch, please make sure you have a clean branch for each development

**Comment by guoqqqi** [comment]
ping @Si-ege

File: web/src/components/Upstream/typings.d.ts [hunk 1]
```diff
@@ -19,7 +19,11 @@
  * Schema: https://github.com/apache/apisix/blob/master/apisix/schema_def.lua#L335
  */
 declare namespace UpstreamComponent {
-  type ActiveCheck = {};
+  type ActiveCheck = {
+    url?: string;
+    host?: string;
+    http_path?: string;
```
Comment (zaunist) [inline web/src/components/Upstream/typings.d.ts:25]: IMO, this is to merge two input boxes into one, right?  so why are there two fields, http_path and host?

Comment (guoqqqi) [reply web/src/components/Upstream/typings.d.ts:25]: I think it may be that, when editing, the interface should still return two fields 🤔

**Comment by guoqqqi** [comment]
also CC @Baoyuantop PTAL

File: web/src/components/Upstream/typings.d.ts (hunk 1 above)
Comment (zaunist) [reply web/src/components/Upstream/typings.d.ts:25]: Make sense.

File: web/src/pages/Upstream/Create.tsx [hunk 2]
```diff
@@ -44,6 +43,15 @@ const Page: React.FC = (props) => {
           }
         }
 
+        if (newData?.checks?.active) {
+          const host = newData?.checks?.active.host;
+          const http_path = newData?.checks?.active.http_path;
+          const url = host + http_path;
+          const { active: activeData } = newData.checks;
+          delete activeData.host;
+          delete activeData.http_path;
```
Comment (Baoyuantop) [inline web/src/pages/Upstream/Create.tsx:52]: We would prefer to use a deep clone before these operations.
I see that the operation in line 37 is not quite suitable, can you help me to change it?

File: web/cypress/integration/upstream/create_and_delete_upstream.spec.js [hunk 3]
```diff
@@ -55,6 +55,7 @@ context('Create and Delete Upstream', () => {
     weight0: '2',
     port1: '7001',
     weight1: '2',
+    url_input: 'www.baidu.com/test/asd',
```
Comment (Baoyuantop) [inline web/cypress/integration/upstream/create_and_delete_upstream.spec.js:58]: Do we need to test for incorrect inputs?

File: web/src/pages/Upstream/Create.tsx (hunk 2 above)
Comment (Si-ege) [reply web/src/pages/Upstream/Create.tsx:52]: I don't quite understand what you mean?

Comment (Baoyuantop) [reply web/src/pages/Upstream/Create.tsx:52]: OK, this is only a suggestion and not a necessity.

File: web/src/pages/Upstream/Create.tsx [hunk 4]
```diff
@@ -44,6 +44,14 @@ const Page: React.FC = (props) => {
           }
         }
 
+        if (newData?.checks?.active) {
+          const host = newData?.checks?.active.host;
+          const http_path = newData?.checks?.active.http_path;
+          const url = host + http_path;
+          newData.checks.active.url = url;
```
Comment (Baoyuantop) [inline web/src/pages/Upstream/Create.tsx:51]: Better to use template literals.

Comment (guoqqqi) [reply web/src/pages/Upstream/Create.tsx:51]: means using this syntax
image

**Comment by guoqqqi** [comment]
It seems that the front-end CI is unstable and we can commit an empty commit to trigger the CI 
https://www.freecodecamp.org/news/how-to-push-an-empty-commit-with-git/

File: web/src/pages/Upstream/Create.tsx [hunk 5]
```diff
@@ -44,6 +44,15 @@ const Page: React.FC = (props) => {
           }
         }
 
+        if (newData?.checks?.active) {
+          const host = newData?.checks?.active.host;
+          const http_path = newData?.checks?.active.http_path;
+          const url = `${host}${http_path}`;
+          const { active: activeData } = newData.checks;
+          activeData.url = url;
+          newData.checks.active = omit(newData.checks.active, 'host');
```
Comment (guoqqqi) [inline web/src/pages/Upstream/Create.tsx:53]: cc @hanzhenfang PTAL

**Pull request closed without merging**